assembl/graphql/schema.py

```diff
@@ -189,6 +190,9 @@ def resolve_locales(self, args, context, info):
     def resolve_resources_center(self, args, context, info):
         return ResourcesCenter()
 
+    def resolve_visits_analytics(self, args, context, info):
+        return VisitsAnalytics()
```

Comment:
If you want to just do one piwik query instead of three, do the query here and create `VisitsAnalytics(sum_visits_length=sum_visits_length, nb_pageviews=nb_pageviews, nb_uniq_pageviews=nb_uniq_pageviews)`
We do something similar for `Video` in `graphql/idea.py`

Reply:
If you want to know if the graphql query is asking for the fields, you can use `get_fields`, there is an example in `graphql/idea.py`, probably like this:
```
fields = get_fields(info)
if 'sumVisitsLength' in fields:
    ....
```

Reply:
improvement done in https://github.com/assembl/assembl/pull/246/commits/295b190aeffffc3eff225453d6d338c5a28566d8

Reply:
improvement done in 301fd66e4c948aafcca0efe9427c876ef721e8d6 . I tested it by counting printed statements, which worked, but I could not put it in an automated test because we use mock for python 2.7 which does not have `mock_method.assert_called_once()`